infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java

```diff
@@ -34,26 +51,51 @@ public void sealServerWithEpoch(long epoch) {
         // Overridden in log unit to flush operations stamped with an old epoch
     }
 
+    @Deprecated
     public abstract boolean isServerReadyToHandleMsg(CorfuMsg msg);
 
+    /**
+     * Determine if the server is ready to handle a request.
+     * @param request The incoming request message.
+     * @return True if the server is ready to handle this request, and false otherwise.
+     */
+    public boolean isServerReadyToHandleReq(RequestMsg request) {
+        //TODO: Make abstract once other servers are implemented
+        return false;
+    }
+
     /**
      * A stub that handlers can override to manage their threading, otherwise
      * the requests will be executed on the IO threads
      * @param msg
      * @param ctx
      * @param r
      */
+    @Deprecated
     protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         getHandler().handle(msg, ctx, r);
     }
 
+    /**
+     * A stub that handlers can override to manage their threading, otherwise
+     * the requests will be executed on the IO threads
+     * @param req An incoming request message.
+     * @param ctx The channel handler context.
+     * @param r The router that took in the request.
+     */
+    protected void processRequest(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        getHandlerMethods().handle(req, ctx, r);
+    }
```

Comment:
Change this to abstract. The default implementation is not used anywhere, force concrete classes to handle the invocation (that way they can also manage their own thread dispatching).

Reply:
Done.